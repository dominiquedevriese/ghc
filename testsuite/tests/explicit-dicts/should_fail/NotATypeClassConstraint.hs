{-# LANGUAGE ConstraintKinds #-}
module NotATypeClassConstraint where

import Data.Proxy
import GHC.Classes (Eq(..))

f :: c => Proxy c -> ()
f _ = ()

g = f ((eqInt :: c)) Proxy
  where
    eqInt :: Eq.Dict Int
    eqInt = Eq.Dict (==) (/=)
