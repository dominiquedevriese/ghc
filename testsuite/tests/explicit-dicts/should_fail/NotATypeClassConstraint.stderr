
NotATypeClassConstraint.hs:10:5: error:
    • Not a type-class constraint: forall (c :: Constraint). c
    • In the expression: f ((eqInt :: c)) Proxy
      In an equation for ‘g’:
          g = f ((eqInt :: c)) Proxy
            where
                eqInt :: Eq.Dict Int
                eqInt = Eq.Dict (==) (/=)
